### PR TITLE
Heuristics implementation

### PR DESCRIPTION
**Pull request by moratodpg**
This is an initial implementation for computing heuristic decision rules.
We have to discuss the arrangement of our original struct_env: shall we slightly modify it or just inherit it here?
In this case, we need to retrieve inspection observations (detected, no-detected).

### CODE REVIEW

File: heuristics/struct_env_heur.py [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+            self.alphas = alpha_prime
+
+        self.beliefs = belief_prime
+        self.d_rate = drate_prime
+
+        # An episode is done if the agent has reached the target
+        done = self.time_step >= self.ep_length
+
+        # info = {"belief": self.beliefs}
+        return self.observations, rewards, done, observation_
```
Comment (PaLeroy) [inline heuristics/struct_env_heur.py]: If I am correct, this is the single difference

**Comment by PaLeroy** [comment]
For the struct_env_heurisitc, either we inherit from struct env and we only modify the step function or we can modify the step function of struct_env directly and check its use in the other classes.

**Comment by PaLeroy** [comment]
I am ok with this one

**Review by moratodpg — COMMENTED** [review]
Great, thanks for completing the implementation.
I very much like it.

**Pull request merged**